app/models/relationship.rb

```diff
@@ -24,8 +24,10 @@ class Relationship < ApplicationRecord
   after_create :point_targets_to_new_source, :update_counters, prepend: true
   after_update :reset_counters, prepend: true
   after_update :propagate_inversion
+  after_save :set_unmatched_field, if: proc { |r| r.is_confirmed? }
```

Comment:
For clarity, I suggest we rename `set_unmatched_field` to `turn_off_unmatched_field`